### PR TITLE
Replace ROS_* logging macros and use RCLCPP_* instead

### PR DESCRIPTION
**Pull request by ivanpauno**
Trivial fix of old TODO.

### CODE REVIEW

**Review by tfoote — APPROVED** [review]
Glad to see this cleanup now that the logging is available.

**Comment by ivanpauno** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13697/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8574/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11413/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13768/)

**Comment by ivanpauno** [comment]
Again after some fixes:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13698/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8575/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11414/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13769/)

**Comment by clalancette** [comment]
When I tried to fix this in the past, I vaguely remember that there were some codepaths that we could get into without a `node` attached.  You may have fixed that here, but can you re-run CI with `--packages-above tf2_ros` just to be sure?

**Comment by ivanpauno** [comment]
> When I tried to fix this in the past, I vaguely remember that there were some codepaths that we could get into without a node attached. You may have fixed that here, but can you re-run CI with --packages-above tf2_ros just to be sure?

Yes, there was one place were I detected that and it was fixed in https://github.com/ros2/geometry2/pull/380/commits/7904da672f032879eb0bc2d03ff91cd439db499f.

CI running tests above `tf2_ros` to double check that fix was enough: 

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13706/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8580/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11419/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13776/)

**Comment by clalancette** [comment]
> CI running tests above `tf2_ros` to double check that fix was enough:

Thanks!

**Comment by ivanpauno** [comment]
Windows failure is unrelated, going in!

File: tf2_ros/include/tf2_ros/transform_listener.h [hunk 1]
```diff
@@ -71,6 +71,7 @@ class TransformListener
   : buffer_(buffer)
   {
     init(node, spin_thread, qos, static_qos, options);
+    node_logging_interface_ = node->get_node_logging_interface();
```
Comment (jacobperron) [inline tf2_ros/include/tf2_ros/transform_listener.h:74]: `node_logging_interface_` was already initialized in `init()` above

**Comment by jacobperron** [comment]
It looks like this change caused a regression in rviz https://github.com/ros2/rviz/issues/665

I'm not sure if it's a bug here or in rviz yet.

File: tf2_ros/src/buffer.cpp [hunk 2]
```diff
@@ -107,10 +100,10 @@ void Buffer::onTimeJump(const struct rcl_time_jump_t & time_jump)
   if (RCL_ROS_TIME_ACTIVATED == time_jump.clock_change ||
     RCL_ROS_TIME_DEACTIVATED == time_jump.clock_change)
   {
-    ROS_WARN("Detected time source change. Clearing TF buffer.");
+    RCLCPP_WARN(node_->get_logger(), "Detected time source change. Clearing TF buffer.");
```
Comment (jacobperron) [inline tf2_ros/src/buffer.cpp:103]: This is a bug. We're assuming that `node_` is not `nullptr`, but it is actually the default value. The node argument to `tf2_ros::Buffer` is optional. I'll look into a patch and add some test coverage for this code path.

Comment (jacobperron) [reply tf2_ros/src/buffer.cpp:103]: Fixed in https://github.com/ros2/geometry2/pull/393

**Pull request merged**